pkg/controller/autoscaler/horizontalpodautoscaler_controller.go

```diff
@@ -18,23 +18,36 @@ package autoscalercontroller
 
 import (
 	"fmt"
+	"strings"
 	"time"
 
 	"github.com/golang/glog"
 	"k8s.io/kubernetes/pkg/api"
-	"k8s.io/kubernetes/pkg/client/unversioned"
+	client "k8s.io/kubernetes/pkg/client/unversioned"
 	"k8s.io/kubernetes/pkg/fields"
 	"k8s.io/kubernetes/pkg/labels"
 	"k8s.io/kubernetes/pkg/util"
 )
 
+const (
+	heapsterNamespace = "kube-system"
+	heapsterService   = "monitoring-heapster"
+)
+
+var resourceToMetric = map[api.ResourceName]string{
+	api.ResourceCPU: "cpu-usage",
+}
+var heapsterQueryStart, _ = time.ParseDuration("-20m")
+
 type HorizontalPodAutoscalerController struct {
-	kubeClient unversioned.ExperimentalInterface
+	client    *client.Client
```

Comment:
I would love to but at this moment Interface doesn't support calling service via proxy (or I don't know how to do it).